### ISSUE TITLE
man.vim: error in syntax file

### ISSUE THREAD

**Issue by martinhj** [neovim/neovim#5574]
- `nvim --version`:
NVIM 0.2.0-dev
Build type: Dev
(Homebrew development version installed as described at https://github.com/neovim/homebrew-neovim/blob/master/README.md)

- Vim (version: VIM - Vi IMproved 7.3) behaves differently?
Different version of the syntax file in question, so yes.

- Operating system/version:
Mac OS X 10.11.6

- Terminal name/version: na
- `$TERM`: xterm-256color

### Actual behavior
set $MANPAGER  to "nvim -c 'set ft=man' -" as described in ':help vim '. When running for example ' man ls ' neovim gives this error message:

```
Error detected while processing /usr/local/Cellar/neovim/HEAD-32d9c19/share/nvim/ru
ntime/syntax/man.vim:
line   21:
E121: Undefined variable: b:man_sect
E15: Invalid expression: b:man_sect =~# '^[23]'
Press ENTER or type command to continue
```

When running :echo b:man_sect in neovim it returns 1. If I set the environment variable MANSECT to for example 2 and running man ls again :echo b:man_sect returns this. But the b:man_sect variable does not seem to be set when the syntax/man.vim file is parsed and ran anyway.

When I remove the line 'syntax on' from my init.vim file the error is no longer present.

Also if I encapsulates the if b:man_sect =~# conditional in syntax/man.vim file with
```
if exists("b:man_sect") 
```
the problem also disappears.

### Expected behavior
neovim starts without an error message.

### Steps to reproduce using `nvim -u NORC`
When using MANPAGER="nvim -u NORC -c 'set ft=man' - " the problem is not present.

**Comment by justinmk** [comment]
cc @nhooyr

**Comment by nhooyr** [comment]
Something fishy is going on in https://github.com/neovim/neovim/blob/master/runtime/ftplugin/man.vim#L11

Could you add a `echo` there to see if that section is even being ran?

**Comment by nhooyr** [comment]
Actually, since `echo b:man_sect` after results in `1` I think it is safe to assume that it is being run, but after the syntax file. I didn't know that was possible.

**Comment by nhooyr** [comment]
Can you reproduce with `export MANPAGER="nvim -u NORC -c 'set ft=man' -"`?

**Comment by nhooyr** [comment]
Oh, you already tried that. I'm not sure what it could be then. Could you try bisecting your init.vim? E.g, comment out half of it and see if the problem occurs. If it does not, comment out half of the half you commented out before and so on until it does happen.

**Comment by martinhj** [comment]
I already tried that also. If I leave out 'syntax on' it runs without errors, nothing else matters (tried all commented out with only 'syntax on' and also only commented out 'syntax on').

**Comment by nhooyr** [comment]
I've managed to reproduce thanks. I'll have a fix PR in a few hours.

**Comment by nhooyr** [comment]
Can't you just get rid of `syntax on` in your init.vim? There is no need for it in neovim.

**Comment by nhooyr** [comment]
I fixed it anyway in #5630.
